server/sonar-server/src/main/java/org/sonar/server/qualitygate/QualityGateFinder.java

```diff
@@ -57,20 +60,12 @@ public QualityGateFinder(DbClient dbClient) {
     }
   }
 
-  public QualityGateDto getById(DbSession dbSession, long qualityGateId) {
-    return checkFound(dbClient.qualityGateDao().selectById(dbSession, qualityGateId), "No quality gate has been found for id %s", qualityGateId);
-  }
-
-  public QualityGateDto getByNameOrId(DbSession dbSession, @Nullable String name, @Nullable Long id) {
-    if (name != null) {
-      return checkFound(dbClient.qualityGateDao().selectByName(dbSession, name), "No quality gate has been found for name %s", name);
-    }
-    if (id != null) {
-      return getById(dbSession, id);
-    }
-    throw new IllegalArgumentException("No parameter has been set to identify a quality gate");
+  public QGateWithOrgDto getByOrganizationAndId(DbSession dbSession, OrganizationDto organization, long qualityGateId) {
+    return checkFound(dbClient.qualityGateDao().selectByOrganizationAndId(dbSession, organization, qualityGateId),
+      "No quality gate has been found for id %s in organization %s", qualityGateId, organization.getName());
   }
 
+  // TODO As there is always a default quality gate, this method should not return an optional
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1135)